stumpy/core.py

```diff
@@ -2254,14 +2375,29 @@ def rolling_isconstant(a, w):
     w : numpy.ndarray
         The rolling window size
 
+    custom_func : object, default None
+        A custom, user-defined function that returns boolean numpy ndarray that indicate
+        if a subsequence is constant or not. It takes two arguments, `a`, a 1-D array,
+        and `w`, the window size, and may have default arguments if needed. When `None`,
+        this will be default to the function `core._rolling_isconstant`.
+
     Returns
     -------
     output : numpy.ndarray
         Rolling window isconstant.
     """
+    rolling_isconstant_func = _rolling_isconstant
+    if custom_func is not None:
+        custom_func_args = set(inspect.signature(custom_func).parameters.keys())
+        if len(set(["a", "w"]).difference(custom_func_args)):  # pragma: no cover
+            msg = "Incompatible parameters found in `custom_func`"
```

Comment:
We should probably tell the user what the problem is and how to fix it:

```
incomp_args = list(custom_func_args.difference(set("a", "w")))
msg = f"Incompatible arguments {incomp_args} found in `custom_func`. "
msg += "Please provide a `custom_func` with arguments `a`, a 1-D array, and `w`, the window size. "
msg += "The `custom_func` has been ignored and the default `core._rolling_iscontant` function has been applied."
```

Reply:
@seanlaw 
Sure. Two things:
(1) Just to make sure we are on the same page, I think we should do: `set("a", "w").difference(custom_func_args)`. right?

(2) Since the user is (probably) advanced and wants to use a `custom_func`, shouldn't we just throw an error instead of warning? I am just curious and want to understand how you see this situation.

Reply:
> 1) Just to make sure we are on the same page, I think we should do: set("a", "w").difference(custom_func_args). right?

No, because I actually want to warn the user about the EXTRA unrecognized arguments.

> (2) Since the user is (probably) advanced and wants to use a custom_func, shouldn't we just throw an error instead of warning? I am just curious and want to understand how you see this situation.

Yes, you should throw an ValueError